### PR TITLE
#19652 Ability to query related content in GraphQL

### PR DESCRIPTION
**Pull request by dsilvam**
This PR introduces the ability to use the following parameters when requesting related content in GraphQL:
* query: relatedFieldVariable.fieldName
* sortBy: variable name of the field to sort by
* limit: maximum number results 
* offset: starting rownum

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java [hunk 1]
```diff
@@ -55,8 +56,29 @@ public Object get(final DataFetchingEnvironment environment) throws Exception {
 
             Object objectToReturn = records.doesAllowOnlyOne() ? null : Collections.emptyList();
 
-            List<Contentlet> relatedContent = contentlet.getRelated(fieldVar, user,
-                    true, isChildField, contentlet.getLanguageId(), contentlet.isLive());
+//            List<Contentlet> relatedContent = contentlet.getRelated(fieldVar, user,
+//                    true, isChildField, contentlet.getLanguageId(), contentlet.isLive());
+
+            final String query = environment.getArgument("query")!=null
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java:62]: You can use this apache commons 3 function and code will be cleaner
```
final String query = StringUtils.defaultString(environment.getArgument("query"), "");
```

Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java:62]: Same for the rest of String variables.

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtils.java [hunk 2]
```diff
@@ -674,5 +676,34 @@ public static long count(String query, User user, String tmDate) {
         return getPullResults(relationship, contentletIdentifier, condition, limit, offset, sort,
                 user, tmDate, pullParents, -1, null);
     }
+
+	public static String addDefaultsToQuery(String query, final boolean editOrPreviewMode, final
+			HttpServletRequest request){
+		String q = "";
+
+		if(query != null)
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtils.java:684]: Use a `StringBuilder` instead.

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java [hunk 3]
```diff
@@ -55,8 +56,29 @@ public Object get(final DataFetchingEnvironment environment) throws Exception {
 
             Object objectToReturn = records.doesAllowOnlyOne() ? null : Collections.emptyList();
 
-            List<Contentlet> relatedContent = contentlet.getRelated(fieldVar, user,
-                    true, isChildField, contentlet.getLanguageId(), contentlet.isLive());
+//            List<Contentlet> relatedContent = contentlet.getRelated(fieldVar, user,
```
Comment (erickgonzalez) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java:59]: Remove unused code

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtils.java (hunk 2 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtils.java:684]: agree

**Review by erickgonzalez — CHANGES_REQUESTED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java:62]: I like it! will do

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtils.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtils.java:684]: Java optimizes this under the covers. It's not an advantage anymore.

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java]: Done

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/datafetcher/RelationshipFieldDataFetcher.java]: done

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Pull request merged**